### PR TITLE
locale as query param in GET requests

### PR DESCRIPTION
**Pull request by sarveshpro**
### What to test - 
- check all GET requests if they are working as expected.
- Is locale being passed as a query param.
- other query params shouldn’t be affected.
- Complete Signup flow

### CODE REVIEW

File: src/features/donations/index.tsx [hunk 1]
```diff
@@ -55,8 +55,11 @@ function DonationsPopup({ onClose, project }: Props): ReactElement {
         setIsPaymentOptionsLoading(true);
 
         const paymentSetupData = await getRequest(
-          `/app/projects/${project.id}/paymentOptions?country=${country}`,
-          handleError
+          `/app/projects/${project.id}/paymentOptions`,
+          handleError,
+          {
+            country: country,
+          }
```
Comment (ankitecd) [inline src/features/donations/index.tsx:62]: The `getRequest` function accepts 3rd param as redirect and 4th param as query

File: src/features/donations/legacyDonation.tsx [hunk 2]
```diff
@@ -55,8 +55,11 @@ function LegacyDonations({ paymentData }: Props): ReactElement {
         setIsPaymentOptionsLoading(true);
 
         const paymentSetupData = await getRequest(
-          `/app/projects/${paymentData.plantProjectGuid}/paymentOptions?country=${country}`,
-          handleError
+          `/app/projects/${paymentData.plantProjectGuid}/paymentOptions`,
+          handleError,
+          {
+            country: country,
+          }
```
Comment (ankitecd) [inline src/features/donations/legacyDonation.tsx:62]: Here as well

> The `getRequest` function accepts 3rd param as redirect and 4th param as query

File: src/tenants/salesforce/Home/components/SeaOfTrees.tsx [hunk 3]
```diff
@@ -17,8 +17,12 @@ export default function SeaOfTrees() {
...
       const currencyCode = getStoredCurrency();
       const project = await getRequest(
-        `/app/projects/${projectID}?_scope=extended&currency=${currencyCode}`,
-        handleError
+        `/app/projects/${projectID}`,
+        handleError,
+        {
+          _scope: 'extended',
+          currency: currencyCode,
+        }
```
Comment (ankitecd) [inline src/tenants/salesforce/Home/components/SeaOfTrees.tsx:25]: Here as well

> The `getRequest` function accepts 3rd param as redirect and 4th param as query

File: src/features/donations/index.tsx (hunk 1 above)
Comment (norbertschuler) [reply src/features/donations/index.tsx]: I added '/' now.

File: src/tenants/salesforce/Home/components/SeaOfTrees.tsx (hunk 3 above)
Comment (norbertschuler) [reply src/tenants/salesforce/Home/components/SeaOfTrees.tsx]: I added '/' now.

File: src/features/donations/legacyDonation.tsx (hunk 2 above)
Comment (norbertschuler) [reply src/features/donations/legacyDonation.tsx]: I added '/' now.

**Comment by sagararyal** [comment]
#### 📝 Changed routes:


| [**/\[p\]**](https://planet-webapp-m3780srs8-planetapp.vercel.app/[p]) | [**/**](https://planet-webapp-m3780srs8-planetapp.vercel.app/) |
|:-:|:-:|
|<a href="https://planet-webapp-m3780srs8-planetapp.vercel.app/[p]">Screenshot of /[p]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/tuDKk18keocYJmFqfqrnBmcdF-A4YLRWrbC1WB3aYMOSas_i1o81FFOFmuYle1jbmajO6ZPmyHmpOkT3nZp6Xg.png">(view full size)</a> |<a href="https://planet-webapp-m3780srs8-planetapp.vercel.app/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/7bh9OPR5BKuzwM4fpxaa-opMIT8SBlAX3Spdk6eQ5fSsA9AQgj3CYkCtj3iMbXOPiXRg3-b6e7DT8zT_xiVKqA.png">(view full size)</a> |







Commit d7cef9e1727a3c52e0d6f6450fdc1309fc404f8c (https://planet-webapp-m3780srs8-planetapp.vercel.app).

**Comment by norbertschuler** [comment]
What about all the other API calls in https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/develop/src/utils/apiRequests/api.ts - don't they have to be changed and still use the header `x-locale`?

**Comment by norbertschuler** [comment]
Also there are still some requests not abstracted into our [api.ts](https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/develop/src/utils/apiRequests/api.ts) module:

https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/72baf6a049d312fe5b6c0937fac16cc60931800a/pages/donation-payment/%5Bcode%5D.tsx#L24

https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/72baf6a049d312fe5b6c0937fac16cc60931800a/src/features/donations/components/PaymentFunctions.ts#L20

https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/72baf6a049d312fe5b6c0937fac16cc60931800a/src/features/donations/components/PaymentFunctions.ts#L44

https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/72baf6a049d312fe5b6c0937fac16cc60931800a/src/features/user/CompleteSignup/index.tsx#L164

These debts from the past maybe should be changed towards using api.ts as well.

**Comment by sarveshpro** [comment]
> What about all the other API calls in https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/develop/src/utils/apiRequests/api.ts - don't they have to be changed and still use the header `x-locale`?

This change was only required for GET requests as the response is cached if the URL matches. so the translations from backend weren't loading correctly

**Comment by sarveshpro** [comment]
> 

Updated some of these, some of them are having custom headers so skipped them. PaymentFunctions won't be used now as we shifted to "donate.plant-for-the-planet.org" for donations. we will get rid of the code, I am not yet sure if we are still using the legacy-donation route.

**Review by ankitecd — APPROVED** [review]

**Pull request merged**